overlays.nix

```diff
@@ -0,0 +1,20 @@
+final: prev:
+
+let
+  inherit (prev.stdenv) isAarch32 isAarch64 isDarwin;
+
+  darwinSpecific = if isAarch64 then {
+    inherit (prev.darwin.apple_sdk_11_0.frameworks) Accelerate MetalKit;
+  } else {
+    inherit (prev.darwin.apple_sdk.frameworks) Accelerate CoreGraphics CoreVideo;
+  };
+
+  osSpecific = if isDarwin then darwinSpecific else {};
+in
+
+{
+  llama-cpp = prev.callPackage ./llama-cpp.nix osSpecific;
+  llama-cpp-opencl = prev.callPackage ./llama-cpp.nix (osSpecific // { useOpenCL = true; });
+  llama-cpp-cuda = prev.callPackage ./llama-cpp.nix (osSpecific // { useCuda = true; });
+  llama-cpp-rocm = prev.callPackage ./llama-cpp.nix (osSpecific // { useRocm = true; });
```

Comment:
An alternative to `osSpecific // ...` could be to define `llama-cpp = callPackage ...` once and then use `llama-cpp-xxx = final.llama-cpp.override ...`. This has pros and cons, e.g. the users can apply an overlay to `llama-cpp` once and have the effect propagate to all the variants at once. The cost is that it's easier to get into an infinite recursion I guess

Reply:
That con is why I picked this structure. We can always refactor it into a different shape if the pros weigh out.